### PR TITLE
test: add unit test in the xblock

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR adds unit tests in the XBlock

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
You can change the origin to main

**Review by Ian2012 — APPROVED** [review]

**Review by Ian2012 — APPROVED** [review]

**Review by Ian2012 — APPROVED** [review]

**Review by Ian2012 — APPROVED** [review]

File: filesmanager/tests/test_filesmanager.py [hunk 1]
```diff
@@ -0,0 +1,949 @@
...
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(expected_result, response.json)  # pylint: disable=no-member
+
+    def test_get_directories_non_anonymous_user(self):
+        """
+        Check get directories JSON handler for non-anonymous user.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The directories list are returned with the unpublished directory.
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:108]: This should happen if and only if:
- The user is not a student
- We're in the studio edit view
The first check I believe is happening in line 110, is the other one as well?

File: filesmanager/tests/test_filesmanager.py [hunk 2]
```diff
@@ -0,0 +1,949 @@
...
+                        "name": "Unpublished",
+                        "type": "directory",
+                        "path": "Root/Unpublished",
+                        "metadata": {},
+                        "parentId": "root",
+                        "children": [],
+                    }
+                ],
+            },
+        }
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:151]: This is what the default tree looks like. Can we modify it a bit more?

File: filesmanager/tests/test_filesmanager.py [hunk 3]
```diff
@@ -0,0 +1,949 @@
...
+
+        response = self.xblock.delete_content(self.request)
+
+        self.xblock.delete_asset.assert_has_calls(
+            [call(content) for content in contents]
+        )
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(
+            {"status": "success"}, response.json # pylint: disable=no-member
+        )
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:191]: I don't think we'll need this test anymore. But let's leave it as is for now

File: filesmanager/tests/test_filesmanager.py [hunk 4]
```diff
@@ -0,0 +1,949 @@
...
+        self.request.body = json.dumps(data).encode("utf-8")
+
+        response = self.xblock.delete_content(self.request)
+
+        self.xblock.delete_asset.assert_not_called()
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(
+            {"status": "error", "message": "Path not found"},
+            response.json,  # pylint: disable=no-member
+        )
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:210]: Same comment here

File: filesmanager/tests/test_filesmanager.py [hunk 5]
```diff
@@ -0,0 +1,949 @@
...
+                    "metadata": {},
+                    "parentId": "test-directory-id",
+                }
+            ],
+        }
+
+        self.xblock.create_directory(directory, self.xblock.directories["children"])
+
+        self.xblock.upload_file_to_directory.assert_called_once()
+        self.assertIn(directory["path"], self.xblock.content_paths)
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:501]: can we check the new directory is added to the directory children?

File: filesmanager/tests/test_filesmanager.py [hunk 6]
```diff
@@ -0,0 +1,949 @@
...
+            "thumbnail_url",
+        ]
+        content = Mock()
+        content.location = "test-location"
+        content.thumbnail_location = "test-thumbnail_location"
+        content.name = "test-name"
+        content.content_type = "test-content_type"
+        content.length = 100
+        content.get_id.return_value = "test-id"
+
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:713]:
```suggestion
```

File: filesmanager/tests/test_filesmanager.py [hunk 7]
```diff
@@ -0,0 +1,949 @@
...
+        mock_static_content.serialize_asset_key_with_slash.return_value = "asset_url"
+        mock_get_thumbnail_asset_key.return_value = "thumbnail_url"
+        asset = {
+            "contentType": "test-content-type",
+            "length": 100,
+            "_id": "test-id",
+            "asset_key": "test-asset-key",
+            "displayname": "test-display-name",
+        }
+
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:755]:
```suggestion
```

File: filesmanager/tests/test_filesmanager.py [hunk 8]
```diff
@@ -0,0 +1,949 @@
...
+            {
+                "id": "subdirectorycontent",
+                "name": "SubdirectoryContent",
+                "type": "file",
+                "path": "Root/Unpublished/SubdirectoryContent",
+                "metadata": {},
+                "parentId": "unpublished",
+            }
+        )
+
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:861]:
```suggestion
```

File: filesmanager/tests/test_filesmanager.py [hunk 9]
```diff
@@ -0,0 +1,949 @@
...
+    def test_get_content_by_path_root(self):
+        """
+        Check get_content_by_path method when path is "Root".
+
+        Expected result:
+            - The method returns the root directory, None, None.
+        """
+        path = "Root"
+
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:880]:
```suggestion
```

File: filesmanager/tests/test_filesmanager.py [hunk 10]
```diff
@@ -0,0 +1,949 @@
...
+    def test_get_content_by_path_content_exists(self):
+        """
+        Check get_content_by_path method when content exists.
+
+        Expected result:
+            - The method returns the correct content, index, and parent directory.
+        """
+        path = "Root/Unpublished"
+
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:895]:
```suggestion
```

File: filesmanager/tests/test_filesmanager.py [hunk 11]
```diff
@@ -0,0 +1,949 @@
...
+            self.xblock.directories["children"][0],
+            0,
+            self.xblock.directories["children"],
+        )
+
+        result = self.xblock.get_content_by_path(path)
+
+        self.assertEqual(expected_result, result)
+
+    def test_get_content_by_path_content_does_not_exist(self):
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:906]: Thank you so much for all these test cases! I recommend you check out the ddt library. This way we could cover multiple inputs with a single test case. Here are some examples:
https://github.com/eduNEXT/xblock-limesurvey/blob/main/limesurvey/tests/test_limesurvey.py#L348-L352
You don't need to do it for this PR, but you already know next time. Thanks again!

File: filesmanager/tests/test_filesmanager.py [hunk 12]
```diff
@@ -0,0 +1,949 @@
...
+        """
+        Check get_content_by_path method when content does not exist.
+
+        Expected result:
+            - The method returns (None, None, None).
+        """
+        path = "Root/Nonexistent"
+
```
Comment (mariajgrimaldi) [inline filesmanager/tests/test_filesmanager.py:914]:
```suggestion
```

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: filesmanager/tests/test_filesmanager.py (hunk 1 above)
Comment (BryanttV) [reply filesmanager/tests/test_filesmanager.py:108]: Yes, the only way for `self.get_current_user().opt_attrs.get(ATTR_KEY_ANONYMOUS_USER_ID)` to return `None` is for these two conditions to be met

File: filesmanager/tests/test_filesmanager.py (hunk 2 above)
Comment (BryanttV) [reply filesmanager/tests/test_filesmanager.py:151]: Done

File: filesmanager/tests/test_filesmanager.py (hunk 5 above)
Comment (BryanttV) [reply filesmanager/tests/test_filesmanager.py]: Done

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**